test/boost/cql_query_test.cc

```diff
@@ -91,6 +91,32 @@ SEASTAR_TEST_CASE(test_create_table_with_id_statement) {
     });
 }
 
+SEASTAR_TEST_CASE(test_create_twcs_table_no_ttl) {
+    return do_with_cql_env([](cql_test_env& e) {
+        return seastar::async([&e] {
+            // Create a TWCS table with no TTL defined
+            e.execute_cql("UPDATE system.config SET value='warn' WHERE name='restrict_twcs_default_ttl';").get();
+            e.execute_cql("CREATE TABLE tbl (a int, b int, PRIMARY KEY (a)) WITH "
+                          "compaction = {'class': 'TimeWindowCompactionStrategy', "
+                          "'compaction_window_size': '1', 'compaction_window_unit': 'MINUTES'};").get();
+            e.require_table_exists("ks", "tbl").get();
+            // Ensure ALTER TABLE works
+            e.execute_cql("ALTER TABLE tbl WITH default_time_to_live=60;").get();
+            // LiveUpdate and enforce TTL to be defined
+            e.execute_cql("UPDATE system.config SET value='true' WHERE name='restrict_twcs_default_ttl';").get();
+            assert_that_failed(e.execute_cql("CREATE TABLE tbl2 (a int, b int, PRIMARY KEY (a)) WITH "
+                          "compaction = {'class': 'TimeWindowCompactionStrategy', "
+                          "'compaction_window_size': '1', 'compaction_window_unit': 'MINUTES'};"));
```

Comment:
The last line only checked failure, not success.
Please add here also a test that:
1. Creating now tbl2 *with* default_time_to_live=60 will work (the restriction will not prevent it)
2. After that, trying to ALTER TABLE tbl2 to remove the default_time_to_live will fail (if I understand correctly, the new check is performed in any ALTER TABLE, it doesn't matter if default_time_to_live or the compaction strategy is what was changed).